src/graphics/VertexBuffer.h

```diff
@@ -56,19 +56,30 @@ struct VertexBufferDesc {
 	BufferUsage usage;
 };
 
-class Mappable {
+class Mappable : public RefCounted {
 public:
 	virtual ~Mappable() { }
 	virtual void Unmap() = 0;
 
+	inline Uint32 GetSize() const { return m_size; }
+	inline Uint32 GetCapacity() const { return m_capacity; }
+
 protected:
-	Mappable() : m_mapMode(BUFFER_MAP_NONE) { }
+	Mappable(const Uint32 size) : m_mapMode(BUFFER_MAP_NONE), m_size(size), m_capacity(size) { }
```

Comment:
Nit: Would be good to mark this constructor as 'explicit' otherwise Uint32 can be magically silently converted to a Mappable.